powerstore/resource_storagecontainer.go

```diff
@@ -154,7 +163,73 @@ func (r resourceStorageContainer) Read(ctx context.Context, req tfsdk.ReadResour
 
 // Update resource
 func (r resourceStorageContainer) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
+	log.Printf("Started Update")
+
+	// Get plan values
+	var plan models.StorageContainer
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get current state
+	var state models.StorageContainer
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// a worksround
+	// currently a bug on powerstore, not accepting PATCH call for same values
+
+	storageContainerUpdate := &gopowerstore.StorageContainer{}
+
+	if plan.Name.Value != state.Name.Value {
+		storageContainerUpdate.Name = plan.Name.Value
+	}
+
+	if plan.Quota.Value != state.Quota.Value {
+		storageContainerUpdate.Quota = plan.Quota.Value
+	}
+
+	if plan.StorageProtocol.Value != state.StorageProtocol.Value {
+		storageContainerUpdate.StorageProtocol = gopowerstore.StorageContainerStorageProtocolEnum(plan.StorageProtocol.Value)
+	}
+
+	// Get storageContainer ID from state
+	storageContainerID := state.ID.Value
+
+	// Update storageContainer by calling API
+	_, err := r.p.client.PStoreClient.ModifyStorageContainer(context.Background(), storageContainerUpdate, storageContainerID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error updating storageContainer",
+			"Could not update storageContainerID "+storageContainerID+": "+err.Error(),
+		)
+		return
+	}
+
+	// Get StorageContainer Details
+	storageContainerResponse, err := r.p.client.PStoreClient.GetStorageContainer(context.Background(), storageContainerID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error getting storage container after update",
+			"Could not get storage container after update, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	updateStorageContainerState(&state, storageContainerResponse, nil, "Update")
```

Comment:
How about the update of high_water_mark value, did you check its feasibility? since we will have to document the same if not feasible